### PR TITLE
flawz: init at 0.2.0

### PR DESCRIPTION
**Pull request by 0x61nas**
## Description of changes

Create a new package, and add myself to the maintainers list.

[**flawz**](https://github.com/orhun/flawz) is a Terminal User Interface (TUI) written in Rust for browsing the security vulnerabilities (also known as CVEs)

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [x] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/fl/flawz/package.nix [hunk 1]
```diff
@@ -0,0 +1,60 @@
...
+, installShellFiles
+}:
+
+rustPlatform.buildRustPackage rec {
+  pname = "flawz";
+  version = "0.1.0";
+
+  src = fetchFromGitHub {
+    owner = "orhun";
+    repo = "${pname}";
```
Comment (pluiedev) [inline pkgs/by-name/fl/flawz/package.nix:16]:
```suggestion
    repo = "flawz";
```

File: pkgs/by-name/fl/flawz/package.nix [hunk 2]
```diff
@@ -0,0 +1,60 @@
...
+    installShellCompletion \
+      --bash $OUT_DIR/flawz.bash \
+      --fish $OUT_DIR/flawz.fish \
+      --zsh $OUT_DIR/_flawz
+
+    # Clean up temporary directory
+    rm -rf $OUT_DIR
+  '';
+
+  meta = with lib; {
```
Comment (pluiedev) [inline pkgs/by-name/fl/flawz/package.nix:47]: [Avoid `with lib;`](https://github.com/NixOS/nixpkgs/issues/208242)

File: pkgs/by-name/fl/flawz/package.nix [hunk 3]
```diff
@@ -0,0 +1,60 @@
...
+    description = "A Terminal UI for browsing CVEs";
```
Comment (pluiedev) [inline pkgs/by-name/fl/flawz/package.nix:48]:
```suggestion
    description = "Terminal UI for browsing CVEs";
```

File: pkgs/by-name/fl/flawz/package.nix [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+    longDescription = ''
+      flawz is a Terminal User Interface (TUI) for browsing the security vulnerabilities (also known as CVEs).
```
Comment (pluiedev) [inline pkgs/by-name/fl/flawz/package.nix:50]: Does this really add much to the short description?

File: maintainers/maintainer-list.nix [hunk 5]
```diff
@@ -1174,6 +1174,15 @@
...
   };
+  anas = {
+    email = "anas.elgarhy.dev@gmail.com";
+    github = "0x61nas";
+    githubId = 44965145;
+    name = "Anas Elgarhy";
+    keys = [{
+      fingerprint = "E10B D192 9231 08C7 3C35 7EC3 83E0 3DC6 F383 4086";
+    }];
+  };
```
Comment (pluiedev) [inline maintainers/maintainer-list.nix:1185]: Please separate this out to its own commit, titled `maintainers: add anas`

**Review by pluiedev — CHANGES_REQUESTED** [review]

File: pkgs/by-name/fl/flawz/package.nix (hunk 4 above)
Comment (0x61nas) [reply pkgs/by-name/fl/flawz/package.nix:50]: no, i just thought that it's required. i'll delete it thx

File: pkgs/by-name/fl/flawz/package.nix [hunk 6]
```diff
@@ -0,0 +1,57 @@
...
+  '';
+
+  meta = {
+    description = "Terminal UI for browsing CVEs";
+    homepage = "https://github.com/orhun/flawz";
+    changelog = "https://github.com/orhun/flawz/releases/tag/v${version}";
+    license = with lib.licenses; [ mit asl20 ];
+    mainProgram = "${pname}";
+    maintainers = with lib.maintainers; [ anas ];
+    platforms = with lib; platforms.unix ++ platforms.windows;
```
Comment (pluiedev) [inline pkgs/by-name/fl/flawz/package.nix:54]:
```suggestion
    platforms = with lib.platforms; unix ++ windows;
```

File: pkgs/by-name/fl/flawz/package.nix [hunk 7]
```diff
@@ -0,0 +1,57 @@
...
+    # Clean up temporary directory
+    rm -rf $OUT_DIR
+  '';
+
+  meta = {
+    description = "Terminal UI for browsing CVEs";
+    homepage = "https://github.com/orhun/flawz";
+    changelog = "https://github.com/orhun/flawz/releases/tag/v${version}";
+    license = with lib.licenses; [ mit asl20 ];
+    mainProgram = "${pname}";
```
Comment (pluiedev) [inline pkgs/by-name/fl/flawz/package.nix:52]:
```suggestion
    mainProgram = "flawz";
```

**Review by pluiedev — CHANGES_REQUESTED** [review]

**Review by pluiedev — APPROVED** [review]

**Review by Aleksanaa — APPROVED** [review]

File: pkgs/by-name/fl/flawz/package.nix [hunk 8]
```diff
@@ -0,0 +1,57 @@
...
+    repo = "flawz";
+    rev = "v${version}";
+    hash = "sha256-cMieoxyTk+B4UkT/GFemnouvtVlhukmtu5e9S7KjM0k=";
+  };
+
+  cargoHash = "sha256-oz8jA2p2dHHryGbgAEtBPSxQSKQiLHbpfMFdo1LjWos=";
+
+  nativeBuildInputs = [ pkg-config installShellFiles ];
+
+  buildInputs = [ openssl sqlite ];
```
Comment (Aleksanaa) [inline pkgs/by-name/fl/flawz/package.nix:25]: Try adding `lib.optionals stdenv.isDarwin [ darwin.apple_sdk.frameworks.SystemConfiguration ]`

Comment (0x61nas) [reply pkgs/by-name/fl/flawz/package.nix:25]: am a poor guy, i can't test this, but seems reasonable tho.

i see that there a CI's jobs related to `darwin` but it takes a long times to run, and i sorta wanna get this pr merged so i can start package another packages

Comment (Aleksanaa) [reply pkgs/by-name/fl/flawz/package.nix]: Ah, looks like there's also `darwin.apple_sdk.frameworks.CoreServices`... If you don't want to add support for darwin this time, just add `broken = stdenv.isDarwin; # needing some apple_sdk packages` to `meta`.

Comment (0x61nas) [reply pkgs/by-name/fl/flawz/package.nix]: >  just add broken = stdenv.isDarwin; # needing some apple_sdk packages to meta.

i dont have any way to test and see if it'll build on macOS or not, so I guess this's the only option for now (unless someone who have mac appears and offers help)

**Comment by Scrumplex** [comment]
Result of `nixpkgs-review pr 312751` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>2 packages built:</summary>
  <ul>
    <li>flawz</li>
    <li>flawz.man</li>
  </ul>
</details>

**Review by Scrumplex — APPROVED** [review]

**Comment by orhun** [comment]
Hooray! @0x61nas feel free to add instructions to the `flawz` repo as well.

Thanks everyone! 🚀
